crates/ruff_linter/src/rules/perflint/rules/manual_list_comprehension.rs

```diff
@@ -46,15 +50,35 @@ use crate::checkers::ast::Checker;
 #[violation]
 pub struct ManualListComprehension {
     is_async: bool,
+    comprehension_type: Option<ComprehensionType>,
 }
 
 impl Violation for ManualListComprehension {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
     #[derive_message_formats]
     fn message(&self) -> String {
-        let ManualListComprehension { is_async } = self;
+        let ManualListComprehension {
+            is_async,
+            comprehension_type,
+        } = self;
+        let comprehension_str = match comprehension_type {
+            Some(ComprehensionType::Extend) => "extend",
+            Some(ComprehensionType::ListComprehension) => "comprehension",
+            None => "comprehension",
+        };
         match is_async {
-            false => format!("Use a list comprehension to create a transformed list"),
-            true => format!("Use an async list comprehension to create a transformed list"),
+            false => format!("Use a list {comprehension_str} to create a transformed list"),
+            true => format!("Use an async list {comprehension_str} to create a transformed list"),
```

Comment:
The message with `extend` reads a bit off. `Use a list extend to create a transformed list`. It should be something like `Use `list.extend` to append to an existing list`